src/ctap/storage.rs

```diff
@@ -384,7 +384,9 @@ impl PersistentStore {
         let mut pin_properties = [0; 1 + PIN_AUTH_LENGTH];
         pin_properties[0] = pin_code_point_length;
         pin_properties[1..].clone_from_slice(pin_hash);
-        Ok(self.store.insert(key::PIN_PROPERTIES, &pin_properties)?)
+        self.store.insert(key::PIN_PROPERTIES, &pin_properties)?;
+        // If this second transaction fails, you are forced to retry.
+        Ok(self.store.remove(key::FORCE_PIN_CHANGE)?)
```

Comment:
I assume that this is a very rare event, that the second transaction fails after the first succeeded.
From a specification perspective, both are okay. We force the user to change their PIN until `remove(key::FORCE_PIN_CHANGE)`. Which means they'll just need do the PIN change again.

From a user experience perspective, the current implementation has the downside that the user expects the old PIN to be active, since they received an error. And the new PIN actually works. Upside is that they are protected by the stronger PIN earlier. The confusion with this implementation could lock users out that never try the new PIN. Maybe this is reason to change it to be a transaction? WDYT?

Reply:
Yes, I think that's actually a good reason to use a transaction. I didn't think of it. So the alternative would simply be:

```rust
Ok(self.store.transaction(&[
    StoreUpdate::Insert { key: key::PIN_PROPERTIES, value: &pin_properties },
    StoreUpdate::Remove { key: key::FORCE_PIN_CHANGE }
])?)
```

Reply:
FYI your snippet needed a minor change to work, or gets a

> the trait bound `&[u8; 17]: std::borrow::Borrow<[u8]>` is not satisfied

The array reference has to be a slice instead.

Reply:
Oh right yes, I forgot that with generics it doesn't do the conversion from array reference to slice reference.